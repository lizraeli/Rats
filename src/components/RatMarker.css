.rat:hover {
  cursor: pointer;
  transform: scale(1.25);
}
