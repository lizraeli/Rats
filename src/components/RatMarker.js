import React from "react";

import "./RatMarker.css";

/**
 * Info about a rat sighting
 * @typedef {Object} Rat
*/

/**
 * @function RatMarker
 * @param  {Object} rat
 * @param  {string} image
 * @param  {Function} onRatClick
 * @param  {boolean} selected 
 * @return {JSX.Element} 
 */
const RatMarker = ({ rat, image, onRatClick, selected }) => (
  <img
    className={selected ? "rat selected" : "rat"}
    alt=""
    src={image}
    onClick={() => onRatClick(rat)}
  />
);

export default RatMarker;
