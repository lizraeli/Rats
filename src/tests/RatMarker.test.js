import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import RatMarker from "../components/RatMarker";
import ratImageS from "../rat_images/rat_s.png";

it("should render a single image as a Rat Marker", () => {
  const wrapper = shallow(<RatMarker image={ratImageS} />);
  // log the rendered component
  // console.log(wrapper.debug());

  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should render a non-selected Rat Marker without the 'selected' className", () => {
  const wrapper = shallow(<RatMarker image={ratImageS} selected={false} />);
  expect(wrapper.hasClass("selected")).toEqual(false);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should renders a selected Rat Marker with the 'selected' className", () => {
  const wrapper = shallow(<RatMarker image={ratImageS} selected={true} />);
  // console.log(wrapper.debug());
  expect(wrapper.hasClass("selected")).toEqual(true);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
