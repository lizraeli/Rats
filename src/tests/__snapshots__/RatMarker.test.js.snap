// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a non-selected Rat Marker without the 'selected' className 1`] = `
<img
  alt=""
  className="rat"
  onClick={[Function]}
  src="rat_s.png"
/>
`;

exports[`should render a single image as a Rat Marker 1`] = `
<img
  alt=""
  className="rat"
  onClick={[Function]}
  src="rat_s.png"
/>
`;

exports[`should renders a selected Rat Marker with the 'selected' className 1`] = `
<img
  alt=""
  className="rat selected"
  onClick={[Function]}
  src="rat_s.png"
/>
`;
